Sources/Host/POSIX/FileIO.cpp

```diff
@@ -37,6 +38,29 @@ ErrorCode FileIO::CloseFile(int fd) {
   return kSuccess;
 }
 
+ErrorCode FileIO::FileRead(int fd, std::string &buf, uint64_t &count,
+                           uint64_t offset) {
+  if (offset > std::numeric_limits<off_t>::max()) {
+    return kErrorInvalidArgument;
+  }
+
+  off_t offArg = static_cast<off_t>(offset);
+  size_t countArg = static_cast<size_t>(count);
+
+  buf.resize(countArg);
+
+  ssize_t nRead = ::pread(fd, &buf[0], countArg, offArg);
```

Comment:
buf->c_str() is `const` so the compiler will complain.

That being said, using `std::string` for this is kind of crap I think.
`U8Vector` (which should be renamed `ByteVector`) would be more
appropriate, but I know we're not the best at that elsewhere in the project.

On Wed, May 25, 2016 at 3:04 PM, kokarez <notifications@github.com> wrote:

> In Sources/Host/POSIX/FileIO.cpp
> <https://github.com/facebook/ds2/pull/353#discussion_r64661704>:
>
> > @@ -37,6 +38,29 @@ ErrorCode FileIO::CloseFile(int fd) {
> >    return kSuccess;
> >  }
> >
> > +ErrorCode FileIO::FileRead(int fd, std::string &buf, uint64_t &count,
> > +                           uint64_t offset) {
> > +  if (offset > std::numeric_limits<off_t>::max()) {
> > +    return kErrorInvalidArgument;
> > +  }
> > +
> > +  off_t offArg = static_cast<off_t>(offset);
> > +  size_t countArg = static_cast<size_t>(count);
> > +
> > +  buf.resize(countArg);
> > +
> > +  ssize_t nRead = ::pread(fd, &buf[0], countArg, offArg);
>
> why not buf->c_str() ?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/353/files/00af9f8f51ac68b234e2901711792e11cf956058..56b2e8ce70a38350cc279f43e0a89254ee8209d9#r64661704>
>



-- 
Stephane Sezer